plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -65,26 +66,62 @@ export class TeamManager {
             maxAge: ONE_HOUR * 24, // cache up to 24h
             updateAgeOnGet: true,
         })
+        this.tokenToTeamIdCache = new LRU({
+            max: 100_000,
+        })
         this.propertyDefinitionsCache = new PropertyDefinitionsCache(serverConfig, statsd)
         this.instanceSiteUrl = serverConfig.SITE_URL || 'unknown'
     }
 
     public async fetchTeam(teamId: number): Promise<Team | null> {
         const cachedTeam = this.teamCache.get(teamId)
-        if (cachedTeam) {
+        if (cachedTeam !== undefined) {
             return cachedTeam
         }
 
         const timeout = timeoutGuard(`Still running "fetchTeam". Timeout warning after 30 sec!`)
         try {
-            const team: Team | null = (await this.db.fetchTeam(teamId)) || null
+            const team: Team | null = await this.db.fetchTeam(teamId)
             this.teamCache.set(teamId, team)
             return team
         } finally {
             clearTimeout(timeout)
         }
     }
 
+    public async getTeamByToken(token: string): Promise<Team | null> {
+        const cachedTeamId = this.tokenToTeamIdCache.get(token)
+
+        // tokenToTeamIdCache.get returns `undefined` if the value doesn't
+        // exist so we check for the value being `null` as that means we've
+        // explictly cached that the team does not exist
+        if (cachedTeamId === null) {
+            return null
+        } else if (cachedTeamId) {
+            const cachedTeam = this.teamCache.get(cachedTeamId)
+            if (cachedTeam) {
+                return cachedTeam
+            }
+        }
+
+        const timeout = timeoutGuard(`Still running "fetchTeam". Timeout warning after 30 sec!`)
+        try {
+            const team = await this.db.fetchTeamByToken(token)
+            if (!team) {
+                // explicitly cache a null to avoid
+                // unnecessary lookups in the future
+                this.tokenToTeamIdCache.set(token, null)
```

Comment:
Based on how we've messed up in the past here instead of `null` or `undefined` lets instead of those easy to get value instead use some custom string that tells us the team doesn't exist, e.g. create a const wiht a value like `"NO_MATCHING_TEAM_FOUND_FOR_TOKEN_CACHE_VALUE"`

Reply:
Unsure how I feel about mixing types like this. I feel like that's more prone to errors.

Maybe I'm just too used to JS by now that null vs undefined makes some sense to me. `null` is an "empty" value that was explicitly set, whereas `undefined` is a value that's never been set. It is annoying that the pattern in JS is not to throw when accessing something that doesn't exist, but yeah.

Reply:
if you don't want to mix types and this is `int`, then we can use `-42` or something else like that. I do really worry about us messing this up if not in the original pr, then in future changes. Could you make sure we have proper testing on this then minimally.